CMakeLists.txt

```diff
@@ -276,11 +276,12 @@ if (HSA_USE_AMDGPU_BACKEND)
   endif (NOT AMDPHDRS)
 endif ()
 
-set(HSA_AMDGPU_GPU_TARGET "kaveri" CACHE STRING "Target GPU device (kaveri,carrizo,fiji)")
+set(HSA_AMDGPU_GPU_TARGET "" CACHE STRING "Target GPU device (kaveri,carrizo,fiji)")
 
 if (NOT (HSA_AMDGPU_GPU_TARGET STREQUAL "kaveri" OR
          HSA_AMDGPU_GPU_TARGET STREQUAL "carrizo" OR
-         HSA_AMDGPU_GPU_TARGET STREQUAL "fiji"))
+         HSA_AMDGPU_GPU_TARGET STREQUAL "fiji" OR
+         HSA_AMDGPU_GPU_TARGET STREQUAL ""))
```

Comment:
I'm thinking instead of having an empty string to enable auto detection, how about using "auto"